.github/workflows/on_propen_push_do_build.yml

```diff
@@ -0,0 +1,38 @@
+name: On PR Open or Push Do Build
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - 15.[0-9]+.x
+      - 14.0.x
+
+concurrency:
+  # Only cancel jobs for PR updates
+  group: ci-build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Build Infinispan
+        uses: ./.github/actions/build-infinispan
+
+      - run: echo Build Completed ${{ github.run_id }} > server-version.txt
```

Comment:
Do we really need to upload this file. Can't we extract the version of the server based upon the pattern in the server/src path?

Reply:
> Do we really need to upload this file. Can't we extract the version of the server based upon the pattern in the server/src path?
> 
> We could even go further and have a `./github/actions/download-infinispan` that extracts the server and src zips, then sets the server version as an output that can be consumed by subsequent steps. This should reduce the verbosity of some of our workflows and centralise duplicated logic.

do you mean from the file name in the artifact? If it's a reliable info sure we can

Reply:
Yep. That version number is dictated by the project.version, so it's effectively the same as what the action currently does.